lib/ImageCache.js

```diff
@@ -77,16 +77,67 @@ assign(Img.prototype, EventEmitter.prototype, {
 });
 
 var kInstancePoolLength = 300;
-var _instancePool = [];
-
-function getPooledImage (src) {
-  for (var i=0, len=_instancePool.length; i < len; i++) {
-    if (_instancePool[i].getOriginalSrc() === src) {
-      return _instancePool[i];
+var _instancePool = {
+  length: 0,
+  // Keep all the nodes in memory.
+  elements: {
+    
+  },
+  
+  // Push with 0 frequency
+  push: function (hash, data) {
+    this.length++;
+    this.elements[hash] = {
+      hash: hash, // Helps identifying 
+      freq: 0,
+      data: data
+    };
+  },
+  
+  get: function (path) {
+    var element = this.elements[path];
+    
+    if( element ){
+      element.freq++;
+      return element.data;
+    }
+    
+    return null;
+  },
+  
+  // used to explicitely remove the path
+  removeElement: function (path) {
+    // Now almighty GC can claim this soul
+    var element = this.elements[path];
+    delete this.elements[path];
+    this.length--;
+    return element;
+  },
+  
+  _reduceLeastUsed: function (least, currentHash) {
+      var current = this.elements[currentHash];
```

Comment:
Can use `_instancePool.elements[currentHash]` instead of `this`, and remove the runtime binding in `popLeastUsed()`.